app/src/main/java/com/keylesspalace/tusky/fragment/ViewImageFragment.kt

```diff
@@ -131,49 +132,74 @@ class ViewImageFragment : ViewMediaFragment() {
-                    .into(photoView)
-
+    private fun loadImageFromNetwork(url: String, previewUrl: String?, photoView: ImageView) {
+        val glide = Glide.with(this)
+        // Request image from the any cache
+        glide
+                .load(url)
+                .dontAnimate()
+                .onlyRetrieveFromCache(true)
+                .let {
+                    if (previewUrl != null)
+                        it.thumbnail(glide
+                                .load(previewUrl)
+                                .dontAnimate()
+                                .onlyRetrieveFromCache(true)
+                                .centerInside()
+                                .addListener(ImageRequestListener(true, isThumnailRequest = true)))
+                    else it
+                }
+                //Request image from the network on fail load image from cache
+                .error(glide.load(url)
+                        .centerInside()
+                        .addListener(ImageRequestListener(false, isThumnailRequest = false))
+                )
+                .centerInside()
+                .addListener(ImageRequestListener(true, isThumnailRequest = false))
+                .into(photoView)
+    }
 
     /**
      * @param isCacheRequest - is this listener for request image from cache or from the network
      */
-    private inner class ImageRequestListener(private val isCacheRequest: Boolean) : RequestListener<Drawable> {
-        override fun onLoadFailed(e: GlideException?, model: Any?, target: Target<Drawable>?, isFirstResource: Boolean): Boolean {
-            if (isCacheRequest) //Complete the transition on failed image from cache
-                completeTransition()
-            else
-                progressBar?.hide() //Hide progress bar only on fail request from internet
+    private inner class ImageRequestListener(
+            private val isCacheRequest: Boolean,
+            private val isThumnailRequest: Boolean) : RequestListener<Drawable> {
+
+        override fun onLoadFailed(e: GlideException?, model: Any?, target: Target<Drawable>?,
+                                  isFirstResource: Boolean): Boolean {
+            // If cache for full image failed, complete transition
+            if (isCacheRequest && !isThumnailRequest) photoActionsListener.onBringUp()
+            // Hide progress bar only on fail request from internet
+            if (!isCacheRequest) progressBar?.hide()
             return false
         }
 
-        override fun onResourceReady(resource: Drawable?, model: Any?, target: Target<Drawable>?, dataSource: DataSource?, isFirstResource: Boolean): Boolean {
-            progressBar?.hide() //Always hide the progress bar on success
+        override fun onResourceReady(resource: Drawable?, model: Any?, target: Target<Drawable>?,
+                                     dataSource: DataSource?, isFirstResource: Boolean): Boolean {
```

Comment:
I think the params in this override can all be non-null. Makes the subsequent code more readable. Glide does not use `@NonNull` annotations in Java so Kotlin can't pick it up, but they always document nullable stuff.